tags.go

```diff
@@ -200,3 +206,66 @@ func (client *Client) DeleteTag(id ID) error {
 	err := client.Mutate(&m, v, WithName("TagDelete"))
 	return HandleErrors(err, m.Payload.Errors)
 }
+
+// ReconcileTags manages tags API operations for TaggableResourceInterface implementations
+//
+// Tags not in 'tagsWanted' will be deleted, new tags from 'tagsWanted' will be created. Reconciled tags are returned.
+func (client *Client) ReconcileTags(resourceType TaggableResourceInterface, tagsWanted []string) ([]Tag, error) {
+	var err error
+	var tagConnection *TagConnection
+	var assignedTags []Tag
+
+	// Get underlying resource type for type checking
+	switch resource := resourceType.(type) {
+	case *Domain:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *InfrastructureResource:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *Repository:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *Service:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *System:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *Team:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *User:
+		tagConnection, err = resource.GetTags(client, nil)
+	default:
+		return assignedTags, fmt.Errorf("not a taggable resource type: %s" + string(resourceType.ResourceType()))
+	}
+	if err != nil {
+		return assignedTags, err
+	}
+	if tagConnection == nil {
+		return assignedTags, fmt.Errorf("no tags found on %s with id '%s'", string(resourceType.ResourceType()), resourceType.ResourceId())
```

Comment:
Should this return an error? IMO just returning an empty array is fine.

Reply:
Also it's unclear to me if the returned `tagConnection` could be nil (not the fault of this PR)

Reply:
Here we expect `tagConnection` to be a `TagConnection` with a `Nodes` field of type `[]Tag`. This is an extra check that would most likely be covered by the `err` check above. An empty `tagConnection.Nodes` slice is fine but, but `tagConnection` should not be nil. This also gets ahead of a nil pointer dereference in the for loop below

Reply:
Since we are only using `range` [which is nil-safe it will not result in any nil pointer exceptions](https://docs.datadoghq.com/code_analysis/static_analysis_rules/go-best-practices/avoid-nil-check-loop/#description)

Reply:
^This is incorrect since we are accessing `nil.something`